### PR TITLE
Fixed failing precondition for ECRECOVER call

### LINKED ISSUE #1136: 3110546 EC precompile subsection NPE
## Block number

```rust
3110546-3110546
```

## Issue

block `3110546` on `Sepolia` witg 0.5.3-beta
catch by @FlorianHuc 

[ERROR] 2024-09-09 06:42:34.738 [vert.x-worker-thread-3] ZkTracer - First exception that was caught while tracing:
java.lang.IllegalArgumentException: null
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:129) ~[guava-33.0.0-jre.jar:?]
        at net.consensys.linea.zktracer.module.hub.section.call.precompileSubsection.EllipticCurvePrecompileSubsection.resolveAtContextReEntry(EllipticCurvePrecompileSubsection.java:64) ~[?:?]
        at net.consensys.linea.zktracer.module.hub.defer.DeferRegistry.resolveUponContextReEntry(DeferRegistry.java:218) ~[?:?]
        at net.consensys.linea.zktracer.module.hub.Hub.traceContextReEnter(Hub.java:704) ~[?:?]
        at net.consensys.linea.zktracer.ZkTracer.traceContextReEnter(ZkTracer.java:301) ~[?:?]

### PR DESCRIPTION
**Pull request by OlivierBBB**

### CODE REVIEW

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/oob/OobOperation.java [hunk 1]
```diff
@@ -304,19 +302,7 @@ public void populateColumnsForPrecompile(MessageFrame frame) {
...
-    //  decremented by the upfront cost. If not we must replace remainingGas with the
-    //  decremented version
-    long remainingGas = frame.getRemainingGas();
-    long gasCap =
-        Words.unsignedMin(
-            allButOneSixtyFourth(remainingGas), Words.clampedToLong(frame.getStackItem(0)));
-    long callGasLong =
-        transfersValue ? gasCap + ZkTracer.gasCalculator.getAdditionalCallStipend() : gasCap;
-    final BigInteger callGas = BigInteger.valueOf(callGasLong);
+    final BigInteger calleeGas = ((PrecompileCommonOobCall) oobCall).getCalleeGas();
```
Comment (OlivierBBB) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/oob/OobOperation.java]: This here is the point of everything else.

Comment (OlivierBBB) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/oob/OobOperation.java]: We are no longer (falsely) recomputing the gas allowance of the callee. We are getting it like so

```rust
StpCall
  -> CallSection
    -> PrecompileSubsection
      -> EllipticCurvePrecompileSubsection
        -> OobCall
          -> OobOperation
```

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/call/precompileSubsection/EllipticCurvePrecompileSubsection.java [hunk 2]
```diff
@@ -35,12 +35,13 @@ public class EllipticCurvePrecompileSubsection extends PrecompileSubsection {
   public EllipticCurvePrecompileSubsection(Hub hub, CallSection callSection) {
     super(hub, callSection);
 
+    long calleeGas = callSection.stpCall.effectiveChildContextGasAllowance();
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/call/precompileSubsection/EllipticCurvePrecompileSubsection.java]: could be final

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/call/precompileSubsection/BlakeSubsection.java [hunk 3]
```diff
@@ -68,7 +68,8 @@ public BlakeSubsection(Hub hub, CallSection callSection) {
     secondImcFragment = ImcFragment.empty(hub);
     fragments.add(secondImcFragment);
 
-    blake2fParamsOobCall = new Blake2fParamsOobCall();
+    long calleeGas = callSection.stpCall.effectiveChildContextGasAllowance();
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/call/precompileSubsection/BlakeSubsection.java]: final

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/call/precompileSubsection/IdentitySubsection.java [hunk 4]
```diff
@@ -35,7 +35,8 @@ public class IdentitySubsection extends PrecompileSubsection {
   public IdentitySubsection(final Hub hub, final CallSection callSection) {
     super(hub, callSection);
 
-    oobCall = new PrecompileCommonOobCall(OOB_INST_IDENTITY);
+    long calleeGas = callSection.stpCall.effectiveChildContextGasAllowance();
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/call/precompileSubsection/IdentitySubsection.java]: same

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/call/precompileSubsection/ModexpSubsection.java [hunk 5]
```diff
@@ -119,7 +119,8 @@ public ModexpSubsection(final Hub hub, final CallSection callSection) {
 
     final ImcFragment sixthImcFragment = ImcFragment.empty(hub);
     fragments().add(sixthImcFragment);
-    sixthOobCall = new ModexpPricingOobCall();
+    long calleeGas = callSection.stpCall.effectiveChildContextGasAllowance();
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/call/precompileSubsection/ModexpSubsection.java]: itou

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/call/precompileSubsection/ShaTwoOrRipemdSubSection.java [hunk 6]
```diff
@@ -37,10 +37,11 @@ public class ShaTwoOrRipemdSubSection extends PrecompileSubsection {
   public ShaTwoOrRipemdSubSection(Hub hub, CallSection callSection) {
     super(hub, callSection);
 
+    long calleeGas = callSection.stpCall.effectiveChildContextGasAllowance();
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/call/precompileSubsection/ShaTwoOrRipemdSubSection.java]: idem

**Review by letypequividelespoubelles — APPROVED** [review]
LGTM, nice edge case

**Pull request merged**